expression/typeinfer_test.go

```diff
@@ -1004,20 +1004,20 @@ func (s *testInferTypeSuite) createTestCase4CompareFuncs() []typeInferTestCase {
 		{"isnull(c_set        )", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag, 1, 0},
 		{"isnull(c_enum       )", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag, 1, 0},
 
-		{"nullif(c_int_d      , 123)", mysql.TypeLong, charset.CharsetBin, mysql.BinaryFlag, 11, 0}, // TODO: tp should be TypeLonglong
+		{"nullif(c_int_d      , 123)", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag, 11, 0},
```

Comment:
Why is the returned type changed? There is no null column in arguments.

Reply:
`nullif` will be rewritten to `if` expression in `expression_rewritter.go` and will contain a NULL datum. The previous implementation contains some BUG, which has been fixed in this PR.